File: 250215/그래프 탐색/graph-traversal.py


def dfs(node):
    global ans
    for nextNode in graph[node]:
        if not visited[nextNode]:
            ans += 1
            visited[nextNode] = True
            dfs(nextNode)
    return ans

n, m = map(int, input().split())

# Write your code here!
graph = [[] for _ in range(n+1)]
visited = [False for _ in range(n+1)]
for _ in range(m):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)

visited[1] = True
ans = 0
print(dfs(1))

File: 250215/그래프 탐색/graph-traversal.java
import java.util.Scanner;
import java.util.ArrayList;

public class Main {
    public static final int MAX_NUM = 1000;
    public static int n, m;

    public static ArrayList<Integer>[] graph = new ArrayList[MAX_NUM + 1];
    public static boolean[] visited = new boolean[MAX_NUM + 1];
    public static int ans = 0;

    public static void DFS(int node) {
        for(int i = 0; i < graph[node].size(); i++) {
            int curNode = graph[node].get(i);
            if(!visited[curNode]) {
                visited[curNode] = true;
                ans++;
                DFS(curNode);
            }
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        n = sc.nextInt();
        m = sc.nextInt();

        for(int i = 0; i <= n; i++) {
            graph[i] = new ArrayList<>();
        }

        while(m-- > 0) {
            int v1 = sc.nextInt();
            int v2 = sc.nextInt();

            graph[v1].add(v2);
            graph[v2].add(v1);
        }

        visited[1] = true;
        DFS(1);

        System.out.println(ans);
    }
}